onmt/modules/Sequencer.lua

```diff
@@ -26,6 +28,16 @@ end
 function Sequencer:_sharedClone()
   local clone = self.network:clone('weight', 'gradWeight', 'bias', 'gradBias')
 
+  local cloneRNN = onmt.utils.Tensor.findModule(clone, self.rnnClass)
```

Comment:
This is a much better solution! I didn't read the code carefully enough to see this is what was happening. Fixed.